### PR TITLE
examples: tests: port applications to gnrc_netif2

### PR DESCRIPTION
**Pull request by miri64**
I've missed some applications in #7424. This takes care of most of them. I also know of `gnrc_ipv6_ext` but when I fix the compile issue the test itself fails. I have to look into that deeper and then will provide a fix for that as well

This PR is part of the network layer remodelling effort:
PR dependencies

### CODE REVIEW

**Comment by smlng** [comment]
I do get lets of linker errors for missing symbols of netif (old version1) functions, is that to be expected? For instance when compiling `examples/nanocoap_server`. Maybe you need to remove `USEMODULE += auto_init_gnrc_netif` from the Makefiles as well?

**Comment by miri64** [comment]
> I do get lets of linker errors for missing symbols of netif (old version1) functions, is that to be expected? For instance when compiling examples/nanocoap_server. Maybe you need to remove USEMODULE += auto_init_gnrc_netif from the Makefiles as well?

Nope, they are all NDP related (look at the objects that cause those linker errors). They should be fixed in #7456.

**Comment by miri64** [comment]
All except the errors involving the devices added in #7414, #7419, #7462, and #7898 should now be fixed. If not I will provide a further follow-up later.

**Comment by miri64** [comment]
Ok. I searched for all the "failed:" in Murdock's log now. Only remaining fails are in tests, that try to link in `cc110x` code (fixed in #7414) and `nordic_softdevice_ble` code (fixed in #7419), the test applications for `xbee` (fixed in #7898) and `slip` (fixed in #7462), and the unittests due to a bug in `gnrc_mac` (fixed in #8016). Everything else is now accounted for in this PR => Ready for review.

File: tests/gnrc_ipv6_ext/main.c [hunk 1]
```diff
@@ -29,38 +29,40 @@
 #include "net/gnrc/pktbuf.h"
 #include "net/gnrc/netreg.h"
 #include "net/gnrc/netapi.h"
-#include "net/gnrc/netif.h"
+#include "net/gnrc/netif2.h"
 #include "net/gnrc/netif/hdr.h"
+#include "xtimer.h"
```
Comment (bergzand) [inline tests/gnrc_ipv6_ext/main.c]: This include seems to be unecessary.

**Comment by miri64** [comment]
@bergzand can you have a look at this as well, please?

**Comment by bergzand** [comment]
> @bergzand can you have a look at this as well, please?

Just started reviewing, give me a few minutes :)

**Comment by miri64** [comment]
> Just started reviewing, give me a few minutes :)

Sorry didn't mean to put you in a hurry, just wanted to make you aware of this PR :-)

**Comment by bergzand** [comment]
> Sorry didn't mean to put you in a hurry, just wanted to make you aware of this PR :-)

No worries :)

**Comment by miri64** [comment]
Addressed.

**Comment by bergzand** [comment]
Ack, ready to squash.

**Comment by miri64** [comment]
Squashed

**Pull request merged**